File: sol/409.cpp
struct Solution {
    int longestPalindrome(string &s) {
        int freq[52] = {0};
        for (char &c : s){
            ++freq[c - (c < 'a' ? 'A' : ('a' - 26))];
        }
        int total = 0;
        bool hasOdd = false;
        for (int i = 0; i < 52; ++i){
            if (freq[i] & 1){
                hasOdd = true;
                total += freq[i] - 1;
            } else {
                total += freq[i];
            }
        }
        return total + hasOdd;
    }
};

File: sol/409/409.py
from collections import Counter

class Solution:
    def longestPalindrome(self, s: str) -> int:
        c = Counter(s)
        hasOdd = False
        total = 0
        for v in c.values():
            if v & 1:
                hasOdd = True
                total += v - 1
            else:
                total += v
        return total + hasOdd
